lua/tests/helpers.lua

```diff
@@ -116,11 +122,130 @@ local function has_all(_, arguments)
   return true
 end
 
+local prev_buf_max = -1
```

Comment:
This seems quite fragile to me because it adds global mutable state to all infoview-touching test runs. Would really prefer if we avoided doing that, we're already struggling with #94 showing us we have too much mutable state.

Reply:
This is something that you'd "opt-in" to on certain tests, and should be backwards compatible with the way we did things before (see e.g. `infoview/lsp_spec.lua`). ISTM that mutable state is pretty much the only way to get this kind of coverage, and we did have mutable state before, it was just littered throughout the tests themselves rather than centralized in a helper.

Did you perhaps mean #93? I'm pretty sure at this point the best way to handle it would be to:
1. Split up `infoview/run_spec.lua`.
2. Add a feature to `nvim-lua/plenary.nvim` that allows tests to early-abort as soon as an `it()` fails, so only the first error is shown.